projects/addon-editor/directives/tiptap-editor/tiptap-editor.directive.ts

```diff
@@ -35,21 +38,36 @@ export class TuiTiptapEditorDirective {
     }
 
     @Output()
-    readonly valueChange = this.editor.valueChange$;
+    readonly valueChange = this.editor.valueChange$.pipe(this.debounceChanges());
 
     @Output()
-    readonly stateChange = this.editor.stateChange$;
+    readonly stateChange = this.editor.stateChange$.pipe(this.debounceChanges());
 
     constructor(
         @Inject(ElementRef) private readonly el: ElementRef<HTMLElement>,
         @Inject(Renderer2) private readonly renderer: Renderer2,
         @Inject(TuiTiptapEditorService) readonly editor: AbstractTuiEditor,
         @Inject(INITIALIZATION_TIPTAP_CONTAINER) readonly editorContainer: HTMLElement,
         @Inject(TIPTAP_EDITOR) private readonly editorLoaded$: Observable<Editor>,
-        @Self() @Inject(TuiDestroyService) destroy$: TuiDestroyService,
+        @Inject(NgZone) private readonly ngZone: NgZone,
+        @Inject(TUI_EDITOR_OPTIONS) readonly options: TuiEditorOptions,
+        @Self() @Inject(TuiDestroyService) private readonly destroy$: TuiDestroyService,
     ) {
         this.editorLoaded$.pipe(takeUntil(destroy$)).subscribe(() => {
             this.renderer.appendChild(this.el.nativeElement, this.editorContainer);
         });
     }
+
+    private debounceChanges<T>(): UnaryFunction<Observable<T>, Observable<T>> {
+        return pipe(
+            switchMap((value, index) =>
+                index && !!value
+                    ? timer(this.options.emitUpdateTime).pipe(map(() => value))
+                    : of(value),
+            ),
+            tuiZonefree(this.ngZone),
+            distinctUntilChanged(),
+            takeUntil(this.destroy$),
```

Comment:
Probably that we don't need `takeUntil(this.destroy$)` here because `valueChange` and `stateChange` are marked as `@Output()`